ash/src/extensions/khr/video_queue.rs

```diff
@@ -0,0 +1,182 @@
+    pub unsafe fn cmd_end_video_coding(
+        &self,
+        command_buffer: vk::CommandBuffer,
+        end_coding_info: &vk::VideoEndCodingInfoKHR<'_>,
+    ) {
+        (self.fp.cmd_end_video_coding_khr)(command_buffer, end_coding_info);
+    }
+
+    #[inline]
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCreateVideoSessionKHR.html>
+    pub unsafe fn create_video_session(
+        &self,
+        create_info: &vk::VideoSessionCreateInfoKHR<'_>,
+        allocation_callbacks: Option<&vk::AllocationCallbacks<'_>>,
+    ) -> VkResult<vk::VideoSessionKHR> {
+        let mut video_session = mem::zeroed();
+        (self.fp.create_video_session_khr)(
+            self.handle,
+            create_info,
+            allocation_callbacks.as_raw_ptr(),
+            &mut video_session,
+        )
+        .result_with_success(video_session)
+    }
+
+    #[inline]
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkCreateVideoSessionParametersKHR.html>
+    pub unsafe fn create_video_session_parameters(
+        &self,
+        create_info: &vk::VideoSessionParametersCreateInfoKHR<'_>,
+        allocation_callbacks: Option<&vk::AllocationCallbacks<'_>>,
+    ) -> VkResult<vk::VideoSessionParametersKHR> {
+        let mut video_session_parameters = mem::zeroed();
+        (self.fp.create_video_session_parameters_khr)(
+            self.handle,
+            create_info,
+            allocation_callbacks.as_raw_ptr(),
+            &mut video_session_parameters,
+        )
+        .result_with_success(video_session_parameters)
+    }
+
+    #[inline]
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkDestroyVideoSessionKHR.html>
+    pub unsafe fn destroy_video_session(
+        &self,
+        video_session: vk::VideoSessionKHR,
+        allocation_callbacks: Option<&vk::AllocationCallbacks<'_>>,
+    ) {
+        (self.fp.destroy_video_session_khr)(
+            self.handle,
+            video_session,
+            allocation_callbacks.as_raw_ptr(),
+        );
+    }
+
+    #[inline]
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkDestroyVideoSessionParametersKHR.html>
+    pub unsafe fn destroy_video_session_parameters(
+        &self,
+        video_session_parameters: vk::VideoSessionParametersKHR,
+        allocation_callbacks: Option<&vk::AllocationCallbacks<'_>>,
+    ) {
+        (self.fp.destroy_video_session_parameters_khr)(
+            self.handle,
+            video_session_parameters,
+            allocation_callbacks.as_raw_ptr(),
+        );
+    }
+
+    #[inline]
+    /// <https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkGetVideoSessionMemoryRequirementsKHR.html>
+    pub unsafe fn get_video_session_memory_requirements(
+        &self,
+        video_session: vk::VideoSessionKHR,
+    ) -> VkResult<Vec<vk::VideoSessionMemoryRequirementsKHR<'_>>> {
+        read_into_defaulted_vector(|count, data| {
```

Comment:
Same comment regarding `pNext` chains in "returned / written" fields.

Reply:
I'll just note here that `VideoSessionMemoryRequirementsKHR` doesn't have any valid pNext members (but neither did `VideoFormatProperties` when I originally wrote this code).

Reply:
`VkVideoSessionMemoryRequirementsKHR` does have a `pNext` "member"/"field". It may not have any valid `structextends` structures _yet_, but that's the entire point of this API design: allowing extensibility in the _near_ future.

By not dealing with that now, we'll shoot ourselves in the foot and have to provide alternative functions or a semver-breaking release when extensions structures are added. We've had such "broken" abstractions (accidentally) in the past and I'd rather not have them again.

Reply:
Also note that this push-back is annoying and unnecessary.

Reply:
Yup, I understand that! Not objecting to the change, just adding some context.